packages/url-state-provider/src/v2/encode.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import queryString from "query-string"
+import { EncodeOptions, ObjectToEncode, Primitive } from "./types"
+
+const isPrimitive = (value: Primitive | Primitive[]) => {
+  return (
+    typeof value === "string" ||
+    typeof value === "number" ||
+    typeof value === "boolean" ||
+    value === null ||
+    value === undefined ||
+    value instanceof RegExp
+  )
+}
+
+const validateObjectToEncode = (object: ObjectToEncode) => {
+  if (
+    typeof object === "undefined" ||
```

Comment:
we could discuss some tricks to avoid this casting parts using typescript tricks